src/aspenOptimizationLib.py

```diff
@@ -0,0 +1,99 @@
+from dataclasses import dataclass
+import win32com.client as win32
+import sys
+import scipy as sc
+import time
+
+def get_all_children(node):
+    return (node.Elements.Item(i) for i in range(node.Elements.Count))
+    
+def getTEAResult(blocksArray, record_type, searchDict, dataVar):
+    """ Placeholder function for TEA review functionality """
+    aspenOutput = readAspen(blocksArray, record_type, searchDict, dataVar)
+    blocksOutput = list(aspenOutput.keys())
+    totalpower = 0
+    #print(aspenOutput)
+    
+    for blockName in blocksOutput:
+        totalpower += float(aspenOutput[blockName]['Net Power'][0])
+
+    return totalpower
+    
+def aspenBlackBox(valuesArray, isBlock, paramArray, blockNameArray, getTeaResultFunc, blocksArray, dataVar, aspen, record_type, searchDict):
+    if (len(paramArray) != len(blockNameArray) and len(paramArray) != len(valuesArray)):
+        print("ERROR: enter correct number of parameters and blocks")
+        sys.exit()
+        
+    if isBlock == True:
+        typename = "Blocks"
+    else:
+        typename = "Streams"
+    
+    for paramCount in range(len(paramArray)):
+        paramPath = str(blockNameArray[paramCount]) + "\\Input\\" + str(paramArray[paramCount])
+        #print(rf"\Data\{typename}\{paramPath}")
+        paramNode = aspen.Application.Tree.FindNode(rf"\Data\{typename}\{paramPath}")
+        
+        if paramNode is None:
+            print("BAD PATH:", rf"\Data\{typename}\{paramPath}")
+            raise Exception("Node not found")
+        #print(valuesArray[paramCount])
+        paramNode.Value = float(valuesArray[paramCount])
```

Comment:
I think using the `zip` function is nicer here:

```suggestion
    for param, blockName, value in zip(paramArray, blockNameArray, valuesArray):
        paramPath = str(blockName) + "\\Input\\" + str(param)
        # print(rf"\Data\{typename}\{paramPath}")
        paramNode = aspen.Application.Tree.FindNode(rf"\Data\{typename}\{paramPath}")

        if paramNode is None:
            print("BAD PATH:", rf"\Data\{typename}\{paramPath}")
            raise Exception("Node not found")
        # print(valuesArray[paramCount])
        paramNode.Value = float(value)
```